torch/csrc/autograd/variable.h

```diff
@@ -563,15 +508,13 @@ inline Variable make_variable(
     at::Tensor data,
     bool requires_grad = false,
     bool allow_tensor_metadata_change = true) {
-  AT_CHECK(
-      !data.is_variable(),
-      "Must not create a new variable from a variable, use its .data()");
```

Comment:
We remove this check, because there are call sites of `make_variable()` (such as `THPVariable_get_data`) that expects to simply pass in a Variable or non-Variable tensor, and get a Variable with a new autograd history. We could make the `Variable(c10::intrusive_ptr<at::TensorImpl> self)` constructor public, to address the use case of constructing a Variable from a Variable, and avoid removing the check here, but I think the semantics of `make_variable()` here is still clear enough without the check that we should just do it the current way.

Reply:
Well, the reason they now pass in a Variable/non-Variable tensor, is because you deleted the `data()` call that would have turned it into a tensor without autograd history.

Now, I am *sympathetic* to the removal, because it is marginally more efficient; if you keep the `data()` call, you have to do Variable to Tensor to Variable; whereas here only a single shallow copy occurs. So I guess we should keep it as is.

Reply:
As discussed before, I think we should still have `is_variable()` check for now (or whatever is the right thing to check -- that there isn't an autograd_meta?

Reply:
Fixed.